certification/internal/shell/has_unique_tag.go

```diff
@@ -13,31 +11,34 @@ import (
 type HasUniqueTagCheck struct{}
 
 func (p *HasUniqueTagCheck) Validate(image string) (bool, error) {
-	imageName := strings.Split(image, ":")[0]
-	stdouterr, err := exec.Command("skopeo", "list-tags", "docker://"+imageName).CombinedOutput()
+	tags, err := p.getDataToValidate(image)
 	if err != nil {
-		log.Error("unable to execute skopeo on the image: ", err)
 		return false, err
 	}
+	return p.validate(image, tags)
+}
+
+func (p *HasUniqueTagCheck) getDataToValidate(image string) ([]string, error) {
+
+	runReport, err := skopeoEngine.ListTags(image)
 
-	// we must send gojq a []interface{}, so we have to convert our inspect output to that type
-	var skopeoData map[string]interface{}
-	err = json.Unmarshal(stdouterr, &skopeoData)
 	if err != nil {
-		log.Error("unable to parse skopeo list-tags data for image", err)
-		log.Debug("error marshaling skopeo list-tags data: ", err)
-		log.Trace("failure in attempt to convert the raw bytes from `skopeo list-tags` to a [map[string]interface{}")
-		return false, err
+		log.Error("unable to execute skopeo on the image: ", err)
+		return nil, err
 	}
-	tags := skopeoData["Tags"].([]interface{})
 
-	var tagsString string
-	for _, tag := range tags {
-		tagsString = tagsString + tag.(string) + " "
+	return runReport.Tags, nil
+}
+
+func (p *HasUniqueTagCheck) validate(image string, lines []string) (bool, error) {
+
+	tags := ""
+	for _, tag := range lines {
+		tags = tags + tag + " "
 	}
-	log.Debugf(fmt.Sprintf("detected these tags for %s image: %s", imageName, tagsString))
+	log.Debugf(fmt.Sprintf("detected these tags for %s: %s", image, tags))
 
-	if len(tags) > 1 || len(tags) == 1 && strings.ToLower(tags[0].(string)) != "latest" {
+	if len(lines) > 1 || len(lines) == 1 && strings.ToLower(lines[0]) != "latest" {
```

Comment:
Nit: This is fairly opaque. Could you either comment this, or make it more clear implicitly by separating it into small functions with appropriate names, or whatever you think fits best?

Otherwise, I think this looks fine. Marking as approved here so you can merge as soon as you resolve this.

Reply:
Agreed. I added a comment to make it clear how an image passes this check